app/scripts/modules/kubernetes/src/serverGroup/configure/configuration.service.js

```diff
@@ -172,12 +170,20 @@ module.exports = angular.module('spinnaker.serverGroup.configure.kubernetes.conf
           if (matchingContainers.length === 1) {
             validContainers.push(container);
           } else {
-            result.dirty.containers = result.dirty.containers || [];
-            result.dirty.containers.push(container.image);
+            result = result || [];
+            result.push(container.image);
           }
         }
       });
-      command.containers = validContainers;
+      return validContainers;
```

Comment:
the diff here is kind of crappy. i'm not _totally_ happy with this function since it mutates and returns something. i broke it so that it could be used for both container sets. suggestions welcome.

Reply:
Could return `[validContainers, invalidContainers]` as a tuple (not so bad if you use array destructuring) and then mutate `result.dirty` separately.